In [ ]:
###### Estimation and Confidence Intervals:#########

In [3]:
###  a.Build 99% Confidence Interval Using Sample Standard Deviation:#####

In [4]:
import numpy as np
from scipy import stats

In [5]:
data = [1.13, 1.55, 1.43, 0.92, 1.25,
        1.36, 1.32, 0.85, 1.07, 1.48,
        1.20, 1.33, 1.18, 1.22, 1.29]

In [16]:
#Finding Sample Size:
n = len(data)
print("Sample size:", n)

Sample size: 15


In [17]:
#Calculating Sample Mean:
mean = np.mean(data)
print("Sample mean:", round(mean, 4))

Sample mean: 1.2387


In [18]:
#Calculating Sample Standard Deviation:
s = np.std(data, ddof=1)
print("Sample standard deviation:", round(s, 4))

Sample standard deviation: 0.1932


In [19]:
#Finding t Critical Value:
confidence = 0.99
alpha = 1 - confidence

t_value = stats.t.ppf(1 - alpha/2, df=n-1)
print("t critical value:", round(t_value, 4))

t critical value: 2.9768


In [20]:
#Calculating Margin of Error:
margin_error_t = t_value * (s / np.sqrt(n))
print("Margin of error:", round(margin_error_t, 4))

Margin of error: 0.1485


In [21]:
#Calculating 99% confidence interval:
lower_t = mean - margin_error_t
upper_t = mean + margin_error_t

print("99% CI using sample standard deviation:")
print("Lower limit:", round(lower_t, 4))
print("Upper limit:", round(upper_t, 4))


99% CI using sample standard deviation:
Lower limit: 1.0902
Upper limit: 1.3871


###### Steps:
I entered the 15 durability values into Python and calculated the sample mean, sample standard deviation, and sample size. Using a 99% confidence level, I found the t-critical value and calculated the confidence interval as mean ± margin of error. The t-distribution was used because the population standard deviation is unknown and the sample size is small (15). Therefore, the t-distribution provides a more reliable estimate of the true mean durability of the print-heads.


In [ ]:
##### b.Build 99% Confidence Interval Using Known Population Standard Deviation:#######

In [22]:
##Set Population Standard Deviation:
##Given ,sigma=0.2
sigma=0.2

In [26]:
#Finding z-critical value:
z_value = stats.norm.ppf(1 - alpha/2)
print("z critical value:", round(z_value, 4))

z critical value: 2.5758


In [27]:
##Calculating margin of error:
margin_error_z = z_value * (sigma / np.sqrt(n))
print("Margin of error:", round(margin_error_z, 4))

Margin of error: 0.133


In [29]:
##Calculating 99% confidence interval:
lower_z = mean - margin_error_z 
upper_z = mean + margin_error_z 
print("99% CI using known population standard deviation:") 
print("Lower limit:", round(lower_z, 4))
print("Upper limit:", round(upper_z, 4))

99% CI using known population standard deviation:
Lower limit: 1.1057
Upper limit: 1.3717


In [30]:
#######  HYPOTHESIS TESTING #########

In [ ]:
##1.State The Hypothesis#####


μ represent the true mean weekly operating cost of the restaurants.

Null Hypothesis (H₀): The weekly operating cost is equal to the cost predicted by the model.
	   
                                   ​H0: μ=4000

Alternative Hypothesis (H₁): The weekly operating cost is higher than the cost predicted by the model.

                                  ​H1: μ>4000


According to the model, the expected weekly operating cost is

                               W=1000+5(600)=4000

The restaurant owners claim that the actual operating cost is higher, so this is a right-tailed hypothesis test.

In [31]:
#####2.Calculating the Test Statistic:

from scipy.stats import norm

In [32]:
x_bar = 3050
mu = 1000 + 5*600
sigma = 5*25
n = 25

In [34]:
z = (x_bar - mu) / (sigma / (n ** 0.5))
print('Z statistic =', z)

Z statistic = -38.0


In [36]:
######3.Determine the Probability and compare:
z_critical = norm.ppf(1 - alpha)
print('Critical value =', z_critical)

Critical value = 1.6448536269514722


In [37]:
#####4.Making a Decision:
if z > z_critical:
    print('Reject H0')
else:
    print('Fail to Reject H0')

Fail to Reject H0


##### 5.Conclusion:

The weekly operating cost is Rs. 4000, while the sample mean is Rs. 3050. Using a one-tailed z-test with σ = 125 and n = 25, the calculated z-statistic is -38.0. The critical value at α = 0.05 is 1.645. Since -38.0 is less than 1.645, we fail to reject the null hypothesis.There is not enough evidence to support the claim that weekly operating costs are higher than the model suggests.